server/src/main/java/io/kroki/server/service/Plantuml.java

```diff
@@ -65,8 +92,26 @@ public String decode(String encoded) throws DecodeException {
       }
     };
     this.diagramResponse = new DiagramResponse(new Caching(Version.etag()));
-    // Disable include for security reasons
-    OptionFlags.ALLOW_INCLUDE = false;
+    this.includeWhitelist = parseIncludeWhitelist(config.getString("KROKI_PLANTUML_INCLUDE_WHITELIST"));
+    // Disable unsafe include for security reasons
+    OptionFlags.ALLOW_INCLUDE = config.getBoolean("KROKI_PLANTUML_ALLOW_INCLUDE", false);
+    String plantUmlIncludePath = config.getString("KROKI_PLANTUML_INCLUDE_PATH");
+    if (plantUmlIncludePath != null) {
+      // NOTE: relying on classloading sequencing to ensure this gets set before ImportedFiles class is loaded by JVM
+      System.setProperty("plantuml.include.path", plantUmlIncludePath);
+    }
+  }
+
+  private static List<Pattern> parseIncludeWhitelist(String filename) {
+    if (filename != null && new File(filename).isFile()) {
```

Comment:
If the filename is not a file or is not readable I also think we should add a log.